crates/proto/src/h3/h3_client_stream.rs

```diff
@@ -683,6 +701,64 @@ mod tests {
         );
     }
 
+    #[test]
+    fn test_h3_cloudflare_no_grease() {
+        subscribe();
+
+        let cloudflare = SocketAddr::from(([1, 1, 1, 1], 443));
+        let mut request = Message::query();
+        let query = Query::query(Name::from_str("www.example.com.").unwrap(), RecordType::A);
+        request.add_query(query);
+
+        let request = DnsRequest::new(request, DnsRequestOptions::default());
+
+        let mut client_config = client_config();
+        client_config.key_log = Arc::new(KeyLogFile::new());
+
+        let connect = H3ClientStream::builder()
+            .crypto_config(client_config)
+            .build(
+                cloudflare,
+                Arc::from("cloudflare-dns.com"),
+                Arc::from("/dns-query"),
+                true,
+            );
+
+        // tokio runtime stuff...
+        let runtime = Runtime::new().expect("could not start runtime");
+        let mut h3 = runtime.block_on(connect).expect("h3 connect failed");
+
+        let response = runtime
+            .block_on(h3.send_message(request).first_answer())
+            .expect("send_message failed");
+
+        assert!(
+            response
+                .answers()
+                .iter()
+                .any(|record| record.data().as_a().is_some())
+        );
+
+        //
+        // assert that the connection works for a second query
+        // TODO: for some reason AAAA www.example.com. doesnt work here
```

Comment:
It looks like this query is working now

Reply:
For me locally "AAAA www.example.com." still doesnt work against cloudflare when verifying this test works.
This is from the "original" cloudflare test, hence I changed it to "AAAA www.google.com", which does.
To clarify, the comment there just highlights that difference that for some reason "the original query" doesnt work.